lemminx-maven/src/test/java/org/eclipse/lemminx/maven/test/SimpleModelTest.java

```diff
@@ -141,6 +141,24 @@ public void testPropertyHover() throws IOException, InterruptedException, Execut
  		hover = languageService.doHover(document, new Position(15, 13), new XMLHoverSettings());
  		assertNull(hover);
 	}
+	
+	@Test
+ 	public void testPropertyDefinitionSameDocument() throws IOException, InterruptedException, ExecutionException, URISyntaxException {
+		DOMDocument document = createDOMDocument("/pom-with-properties-for-definition.xml", languageService);
+		Position pos = new Position(14, 22);
+		List<? extends LocationLink> actual = languageService.findDefinition(document, pos, () -> {
+		});
+		assertTrue(actual.stream().anyMatch(link -> link.getTargetUri().endsWith("pom-with-properties-for-definition.xml")));
```

Comment:
Note that we aren't testing that the target location is correct, maybe we should?...

Reply:
If it's not too difficult, it would be good to also add the check for it now to detect possible regressions in the future. However, it's not a strong requirement.

Reply:
Done :)